tests/test_keydb.py

```diff
@@ -389,10 +389,16 @@ def test_create_keydb_from_root_metadata(self):
 
     # Ensure only 'keyid2' was added to the keydb database.  'keyid' and
     # 'keyid3' should not be stored.
+    self.maxDiff = None
     self.assertEqual(rsakey2, tuf.keydb.get_key(keyid2))
-    self.assertRaises(tuf.exceptions.UnknownKeyError, tuf.keydb.get_key, keyid)
+
+    test_key = rsakey2
+    test_key['keyid'] = keyid
+    self.assertEqual(test_key, tuf.keydb.get_key(keyid))
```

Comment:
The test is a bit hard to follow here...  Does it really test what the comment above says it tests?